provider/google.go

```diff
@@ -204,12 +204,9 @@ func (p *GoogleProvider) Records() (endpoints []*endpoint.Endpoint, _ error) {
 			if !supportedRecordType(r.Type) {
 				continue
 			}
-			ep := &endpoint.Endpoint{
-				DNSName:    strings.TrimSuffix(r.Name, "."),
-				RecordType: r.Type,
-				Targets:    make(endpoint.Targets, 0, len(r.Rrdatas)),
-				RecordTTL:  endpoint.TTL(r.Ttl),
-			}
+			targets := make(endpoint.Targets, 0, len(r.Rrdatas))
+			ep := endpoint.NewEndpointWithTTL(strings.TrimSuffix(r.Name, "."), r.Type, endpoint.TTL(r.Ttl), targets...)
```

Comment:
https://github.com/kubernetes-incubator/external-dns/blob/1ed5fa6952430456bcf0efb6ce3724bfd6028f8a/endpoint/endpoint.go#L139:6

* `.TrimSuffix` for dnsName is already part of `NewEndpointWithTTL` so we can leave it out here
* we could also remove the subsequent `for` for `r.Rrdatas` by passing in `r.Rrdatas` as the last arguments, refer to the `cleanTargets` stuff in `NewEndpointWithTTL` (if I see that correctly at this time of day).